### PR TITLE
Changed: clean &nbsp; inserted by browser.

### LINKED ISSUE #49: Handle &nbsp; inserted by browser instead of spaces.
When typing, browsers insert `&nbsp;` char whenever it is needed to correctly display inserted space. Since we decided that we want to keep view and model clean, we don't want those `&nbsp;`s. This means, that we need to get clean them.

### PR DESCRIPTION
**Pull request by scofalik**
Fixes #49.

### CODE REVIEW

**Comment by scofalik** [comment]
Changing all `&nbsp;`s from mutated text node may "too much" and it might be incorrect when we introduce "nonbreakable space" feature. Maybe we need more complicated algorithm there, but for now it worked.

File: src/input.js [hunk 1]
```diff
@@ -168,6 +168,17 @@ class MutationHandler {
...
+		// Replace &nbsp; inserted by the browser with normal space.
+		// We want only normal spaces in the model and in the view. Renderer and DOM Converter will be then responsible
+		// for rendering consecutive spaces using &nbsp;, but the model and the view has to be clear.
+		// Other feature may introduce inserting non-breakable space on specific key stroke (for example shift + space).
+		// However then it will be handled outside of mutations, like enter key is.
+		// The replacing is here because it has to be done before `diff` and `diffToChanges` functions, as they
+		// take `newText` and compare it to (cleaned up) view.
+		// It could also be done in mutation observer too, however if any outside plugin would like to
+		// introduce additional events for mutations, they would get already cleaned up version (this may be good or not).
+		mutation.newText = mutation.newText.replace( /\u00A0/g, ' ' );
```
Comment (Reinmar) [inline src/input.js]: You shouldn't modify a property of an object because there can be other listeners interested in this value.

**Comment by Reinmar** [comment]
BTW, you can also create a manual test, similar to the existing typing manual tests, where editor model will be logged every few seconds and `\u00a0` will be replaced there with some visible character (`@` e.g.). The goal will be to see that nbsp never appears in the model regardless of how you use `editor.setData()` or what you type.

**Comment by scofalik** [comment]
Related PR https://github.com/ckeditor/ckeditor5-engine/pull/620

**Comment by scofalik** [comment]
I added a test as suggested by @Reinmar

**Comment by Reinmar** [comment]
While testing this I found just one bug – https://github.com/ckeditor/ckeditor5-typing/issues/52, but it turned out that it's unrelated to this changes. Most likely, it's a FF bug.

Awesome work, @scofalik!

**Pull request merged**